### PR TITLE
Test

### PR DESCRIPTION
**Pull request by Watxesnavo**

### CODE REVIEW

File: src/main/java/org/store/structure/repository/book/BookRepository.java [hunk 1]
```diff
@@ -1,8 +1,13 @@
...
 import org.springframework.data.jpa.repository.JpaRepository;
-import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor;  
+import org.springframework.data.jpa.repository.Query;
 import org.store.structure.model.Book;
 
 public interface BookRepository extends JpaRepository<Book, Long>, JpaSpecificationExecutor<Book> {
+    @Query(value = "SELECT * FROM books b INNER JOIN books_categories bc ON "
+            + "b.id = bc.book_id WHERE bc.category_id = :categoryId", nativeQuery = true)
+    List<Book> findAllByCategoryId(Long categoryId);
```
Comment (Ivan95kos) [inline src/main/java/org/store/structure/repository/book/BookRepository.java:12]: It should work without the custom query
```suggestion
    List<Book> findAllByCategoriesId(Long categoryId);
```

File: src/main/java/org/store/structure/service/book/BookServiceImpl.java [hunk 2]
```diff
@@ -1,35 +1,51 @@
...
 
     @Override
     public BookDto save(CreateBookRequestDto requestDto) {
-        Book book = bookMapper.toModel(requestDto);
+        Book book = bookMapper.toEntity(requestDto);
+        book.setCategories(getCategoriesByIds(requestDto.getCategoryIds()));
         return bookMapper.toDto(bookRepository.save(book));
     }
 
+    private Set<Category> getCategoriesByIds(Collection<Long> ids) {
```
Comment (Ivan95kos) [inline src/main/java/org/store/structure/service/book/BookServiceImpl.java:38]: All private methods should be below public methods

File: src/main/java/org/store/structure/service/book/BookServiceImpl.java [hunk 3]
```diff
@@ -1,35 +1,51 @@
...
+        return ids.stream()
+                .map(categoryRepository::findById)
+                .map(category -> category.orElseThrow(
+                        () -> new EntityNotFoundException("Can't find category " + category)))
+                .collect(Collectors.toSet());
```
Comment (Ivan95kos) [inline src/main/java/org/store/structure/service/book/BookServiceImpl.java:43]: Better to use one DB call than many
try to use `.findAllById(ids)`

File: src/main/java/org/store/structure/service/category/CategoryServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    public CategoryResponseDto save(CategoryRequestDto categoryDto) {
+        Category category = categoryMapper.toEntity(categoryDto);
+        return categoryMapper.toDto(categoryRepository.save(category));
+    }
+
+    @Override
+    @Transactional
+    public CategoryResponseDto update(Long id, CategoryRequestDto categoryDto) {
+        if (categoryRepository.findById(id).isPresent()) {
+            Category category = categoryRepository.findById(id).get();
```
Comment (Ivan95kos) [inline src/main/java/org/store/structure/service/category/CategoryServiceImpl.java:47]: use one call to DB instead of two

**Review by Ivan95kos — CHANGES_REQUESTED** [review]
left a few comments

**Review by kozhukhovsky — APPROVED** [review]

**Pull request merged**